### PR TITLE
fix: wrong position of filter layer when click other column

### PR DESCRIPTION
**Pull request by jajugoguma**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [ ] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description

Fix the issue that showing the filter layer in the wrong position when clicking the filter button of another column

Fix the issue that showing the filter layer in the wrong position when clicking the filter button of another column

**As-Is**
asis

**To-Be**
tobe

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: packages/toast-ui.grid/src/view/filterLayerInner.tsx [hunk 1]
```diff
@@ -66,6 +66,25 @@ export class FilterLayerInnerComp extends Component<Props> {
...
+      const { left } = this.el.getBoundingClientRect();
+      const { clientWidth } = this.el;
+      const { innerWidth } = window;
+      const distance = currentLeft - this.state.left;
+
+      let resultLeft = currentLeft;
+
+      if (innerWidth < left + distance + clientWidth) {
+        resultLeft = currentLeft - (left + distance + clientWidth - innerWidth);
+      }
```
Comment (js87zz) [inline packages/toast-ui.grid/src/view/filterLayerInner.tsx:82]: 이 부분은 주석으로 간단한 설명이 있으면 더 좋겠네요!

File: packages/toast-ui.grid/src/view/filterLayerInner.tsx [hunk 2]
```diff
@@ -66,6 +66,25 @@ export class FilterLayerInnerComp extends Component<Props> {
...
   }
 
+  componentDidUpdate(prevProp: Props) {
+    const { name: currentColName, left: currentLeft } = this.props.columnAddress;
+
+    if (currentColName !== prevProp.columnAddress.name) {
+      const { left } = this.el.getBoundingClientRect();
+      const { clientWidth } = this.el;
+      const { innerWidth } = window;
+      const distance = currentLeft - this.state.left;
```
Comment (js87zz) [inline packages/toast-ui.grid/src/view/filterLayerInner.tsx:76]: `distance` 보다는 차이를 보정하기 위한 거리니까 `diff`(예를 들어)라는 단어를 사용해서 어떤 용도인지 명확히 표현할 수 있으면 좋겠네요~!

**Review by js87zz — APPROVED** [review]
리뷰 완료입니다. 첫 PR 축하드립니다 :)

**Review by adhrinae — APPROVED** [review]
리뷰 완료합니다.
제가 궁금했던 점도 이미 피드백이 있으므로 생략할게요.

**Review by lja1018 — APPROVED** [review]
리뷰완료합니다.

**Pull request merged**